src/h5web/visualizations/line/LineVis.tsx

```diff
@@ -27,36 +33,62 @@ function LineVis(props: Props): ReactElement {
     curveType = CurveType.LineOnly,
     showGrid = true,
     scaleType = ScaleType.Linear,
+    abscissas = range(dataArray.size),
   } = props;
 
-  const indexDomain = extendDomain([0, dataArray.size - 1], 0.01);
+  const abscissaToIndex = getValueToIndexScale(abscissas, true);
+
+  const abscissaDomain = useMemo(() => {
+    const rawDomain = getDataDomain(abscissas);
+    return rawDomain ? extendDomain(rawDomain, 0.01) : undefined;
+  }, [abscissas]);
+
   const dataDomain = useMemo(() => {
     return domain
       ? extendDomain(domain, 0.05, scaleType === ScaleType.Log)
       : DEFAULT_DOMAIN;
   }, [scaleType, domain]);
 
+  if (abscissas.length !== dataArray.size) {
+    throw new Error(
+      `Abscissas size (${abscissas.length}) does not match data length (${dataArray.size})`
+    );
+  }
```

Comment:
A hook can't come after an early return, but I think it can come after a throw. I've just come across this use case in my refactoring and I'm not seeing any linting errors so far. When a component "instance" throws, React probably unmounts it and resets its hook calls history. So you can move this check higher up in the component.